configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -841,15 +841,24 @@ private void moveSearchDefinitionsToSchemasDir(java.nio.file.Path applicationDir
         File schemasDir = applicationDir.resolve(ApplicationPackage.SCHEMAS_DIR.getRelative()).toFile();
         File sdDir = applicationDir.resolve(ApplicationPackage.SEARCH_DEFINITIONS_DIR.getRelative()).toFile();
         if (sdDir.exists() && sdDir.isDirectory()) {
-            File[] sdFiles = sdDir.listFiles();
-            if (sdFiles != null) {
-                Files.createDirectories(schemasDir.toPath());
-                Arrays.asList(sdFiles).forEach(file -> Exceptions.uncheck(
-                        () -> Files.move(file.toPath(),
-                                         schemasDir.toPath().resolve(file.toPath().getFileName()),
-                                         StandardCopyOption.REPLACE_EXISTING)));
+            try {
+                File[] sdFiles = sdDir.listFiles();
+                if (sdFiles != null) {
+                    Files.createDirectories(schemasDir.toPath());
+                    Arrays.asList(sdFiles).forEach(file -> Exceptions.uncheck(
+                            () -> Files.move(file.toPath(),
+                                             schemasDir.toPath().resolve(file.toPath().getFileName()),
+                                             StandardCopyOption.REPLACE_EXISTING)));
+                }
+                Files.delete(sdDir.toPath());
+            } catch (IOException | UncheckedIOException e) {
+                if (schemasDir.exists() && schemasDir.isDirectory())
```

Comment:
Failing if both directories exists would likely break too many applications?

Reply:
Yes, that's the assumption